endpoint/endpoint.go

```diff
@@ -58,6 +58,18 @@ func (ttl TTL) IsConfigured() bool {
 // Targets is a representation of a list of targets for an endpoint.
 type Targets []string
 
+type MXTarget struct {
```

Comment:
Consider adding GoDoc comments to the new MXTarget, SRVTarget, and NewMXTarget for clarity, since these are now part of the public API in the endpoint package.